src/eos-google-chrome

```diff
@@ -94,6 +140,12 @@ class GoogleChromeLauncher:
         except OSError as e:
             exit_with_error("Could not launch Chrome: {}".format(repr(e)))
 
+        self._wait_for_installation()
```

Comment:
We only want to make it the default when it has been installed automatically on the first boot, AFAIK. One thing we could do is to have whatever is being run after the FBE to install Chrome touch a file somewhere (e.g. in XDG_RUNTIME_DIR, for instance) that GS would check after installing Chrome to know whether the installation was user-triggered or not (if it was not, it would be set as the default browser).

Now, thinking a bit more about all this, I'm not sure this script is the right place to encode this new logic at all: the purpose of this script is to show the Chrome icon in the desktop and either run Chrome if installed (keeping the sandbox alive) or open GS to manually install it, and it should be kept as simple as possible while achieving that purpose.

What we need to implement here is different and has little to do with what this helper script is doing: we want some logic that will be run only once after the FBE to automatically start installing Chrome in the background via GS (using the headless mode that needs to be added to GS, IIRC), and have Chrome be set as the default browser afterwards. The only logic in common with the eos-google-chrome script are the bits that check whether the Chrome flatpak is installed, and not much else (or nothing else at all).

So, I'm starting to think we should not add this `--check-only` parameter here because, after all, what we are seeking to achieve by using that new mode of the script is much more than that. Instead, I think we should do something like:
  * Extract the logic checking if the Chrome flatpak is installed to its own module, and use it from eos-google-chrome
  * Create a new script that would be used for the automatic installation on boot (e.g. eos-google-chrome-installer?)
  * Have that script run after the FBE **if** there's an fullly operational internet connection (i.e. Connectivity `full`in `nmcli g status`) only. Otherwise I think it's fine to do nothing as GS would be open if the user clicks on the icon in the desktop.
  * Have eos-google-chrome-installer run GNOME Software in headless-app mode so that Chrome starts installing without any user intervention, and **maybe** monitor that process from the script itself. This would be like the initial approach proposed in this PR but doing it now from a script whose only purpose is to install this in a headless mode, which is more acceptable IMHO because you won't ever run into the situation of running this script twice.
  * Ideally, GNOME Software should make it possible for the script to realize when Chrome has finished installing **in headless-app** mode (e.g. additional `mode` parameter in the ApplicationsChanged signal?), so that it can wait until the process finishes to set it as the default browser.
  * If, for any reason, the user cancels the installation that was automatically triggered at any time, the installer script should notice that too (e.g. another headless-app mode-specific signal?) and the process will simply be aborted (after that point, installing Chrome would require manually intervention).

How does this sound? I think this approach would mean we could keep the logic that sets the browser as the default one outside of GNOME Software, while not artificially modifying the eos-google-chrome script (that is meant to be run each time the user clicks on the icon) to cover the one-off use case we need to implement for the automatic installation feature.